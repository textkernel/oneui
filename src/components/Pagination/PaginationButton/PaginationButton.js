import * as React from 'react';
import PropTypes from 'prop-types';
import { bem } from '../../../utils';
import styles from './PaginationButton.scss';
import { Button } from '../../Buttons';

const { block } = bem('PaginationButton', styles);

export const PaginationButton = (props) => {
    const { children, isActive, ...rest } = props;

    return (
        <Button {...rest} isLink {...block(props)} aria-current={isActive ? 'page' : null}>
            {children}
        </Button>
    );
};

PaginationButton.displayName = 'PaginationButton';

PaginationButton.propTypes = {
    children: PropTypes.number.isRequired,
    isActive: PropTypes.bool,
};

PaginationButton.defaultProps = {
    isActive: false,
};
