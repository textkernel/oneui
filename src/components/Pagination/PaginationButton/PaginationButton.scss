.PaginationButton {
    transition: none;
    border: transparent;
    color: var(--color-neutral-80);

    + .PaginationButton {
        margin-left: var(--space-50);
    }

    &:hover {
        color: var(--color-neutral-80);
        background-color: var(--color-neutral-10);
        text-decoration: none;
    }

    &--isActive {
        color: var(--color-background);
        background-color: var(--color-neutral-60);
        cursor: auto;

        &:hover {
            color: var(--color-background);
            background-color: var(--color-neutral-80);
        }
    }
}
