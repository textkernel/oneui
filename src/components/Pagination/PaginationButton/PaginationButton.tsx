import * as React from 'react';
import { bem } from '../../../utils';
import { Button } from '../../Buttons';
import styles from './PaginationButton.scss';

export interface Props extends React.HTMLAttributes<HTMLElement> {
    children: number;
    isActive?: boolean;
}

const { block } = bem('PaginationButton', styles);

export const PaginationButton: React.FC<Props> = ({ children, isActive = false, ...rest }) => (
    <Button {...rest} isLink {...block({ isActive })} aria-current={isActive ? 'page' : null}>
        {children}
    </Button>
);

PaginationButton.displayName = 'PaginationButton';
