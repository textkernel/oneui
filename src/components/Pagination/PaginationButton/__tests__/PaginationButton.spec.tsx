import React from 'react';
import { render, RenderResult, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { PaginationButton } from '../PaginationButton';

describe('<PaginationButton> that renders a page button', () => {
    let view: RenderResult;

    beforeEach(() => {
        view = render(<PaginationButton data-page={1}>{1}</PaginationButton>);
    });

    it('should render correctly with default props', () => {
        const button = screen.getByRole('button');

        expect(view.container).toMatchSnapshot();
        expect(button).toBeInTheDocument();
        expect(button).not.toHaveAttribute('aria-current');
    });

    it('should add correct classes for active state', () => {
        view.rerender(
            <PaginationButton data-page={1} isActive>
                {1}
            </PaginationButton>
        );
        const button = screen.getByRole('button', { hidden: false });

        expect(button).toBeInTheDocument();
        expect(button).toHaveAttribute('aria-current');
    });
});
