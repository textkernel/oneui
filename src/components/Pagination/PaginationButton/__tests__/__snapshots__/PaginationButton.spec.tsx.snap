// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationButton> that renders a page button should render correctly with default props 1`] = `
<div>
  <button
    class="Button Button--isLink PaginationButton"
    data-page="1"
    type="button"
  >
    1
  </button>
</div>
`;
